### PR TITLE
[WIP]Add HDF file for inputs to linear method matrices

### PR DESCRIPTION
**Pull request by markdewing**
Add the parameter "output_matrix_inputs_hdf" to write out the per-sample values for the wavefunction values, derivatives, Hamiltonian derivatives, etc.   These values are used in `fillOverlapHamiltonianMatrices`.  Along with the file from
"output_linear_matrices_hdf", these files can be used in the `fillFromH5nunit` test.

One set of files is included in this check-in as a regression test. It should be straightforward to create additional sets of files for testing locally, for additional test coverage when making changes to `fillOverlapHamiltonianMatrices`.
(We don't want to increase the size of the repository unnecessarily)

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New feature

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
desktop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes Code added or changed in the PR has been clang-formatted
- Yes This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by ye-luo** [comment]
(paused ALCF CI for troubleshooting)

**Comment by PDoakORNL** [comment]
While the binary files for this are politely small I think we should consider a real solution for the repo and binary files. #3354

File: src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp [hunk 1]
```diff
@@ -724,6 +725,29 @@ QMCCostFunctionBatched::Return_rt QMCCostFunctionBatched::fillOverlapHamiltonian
 {
   ScopedTimer tmp_timer(fill_timer_);
 
+  hdf_archive hout;
+  if (output_matrix_inputs_hdf_)
```
Comment (PDoakORNL) [inline src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp:729]: It seems like when this is set true if I have multiple ranks its going to be a problem.  Am I missing something.

**Review by PDoakORNL — CHANGES_REQUESTED** [review]
Question about feature in multi rank case.

File: src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp (hunk 1 above)
Comment (markdewing) [reply src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp:729]: Yes, it's only for one rank.  I'm not sure how to handle multiple ranks, or if that's a case that's worth handling.

Comment (PDoakORNL) [reply src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp:729]: Just realized if this only used with test programs it should removed from the input and the output argument set to true from the test program. Then some documentation in the code would suffice.

But if its is in the user facing input there must be a fatal error if used on multiple ranks or it needs to do something like put the rank number into the file name and be documented as writing for each rank.

**Comment by markdewing** [comment]
I'm going to mark this as WIP since it probably needs some work on the binary file storage.   The test it enables is targeted at one function.  I can probably move forward using this code/testing locally.

**Comment by markdewing** [comment]
Closing this as some of the testing parts are replaced by #3372.
The commits in this PR will need to be redone to remove the binary files.

**Pull request closed without merging**